src/main/kotlin/org/rust/lang/core/psi/ext/RsPat.kt

```diff
@@ -14,7 +15,31 @@ val RsPat.isIrrefutable: Boolean
         is RsPatTup -> patList.all { it.isIrrefutable }
         is RsPatBox -> pat.isIrrefutable
         is RsPatRef -> pat.isIrrefutable
-        is RsPatStruct -> patFieldList.all { it.pat?.isIrrefutable ?: it.patBinding != null }
+        is RsPatStruct -> patFieldList.all { it.pat?.isIrrefutable ?: (it.patBinding != null) }
         is RsPatConst, is RsPatRange -> false
         else -> true
     }
+
+fun RsPat.skipUnnecessaryTupDown(): RsPat {
+    var pat = this
+    while (pat is RsPatTup) {
+        pat = pat.patList.singleOrNull() ?: return pat
+    }
+    return pat
+}
+
+fun containsStdOptionOrResult(item: RsEnumItem?, patterns: List<RsPat>, exhaustive: Boolean = false): Boolean {
```

Comment:
Hm, the name of this function is very confusing, imo. Also you check `pattern.isIrrefutable` condition twice - inside the function and outside (in `MatchToLetIfIntention`).

I would split this function into 3 small ones:
* for `IfLetToMatchIntention`,
* for `MatchToIfLetIntention`,
* common logic

Reply:
Done. What do you think now?

Reply:
OK